### DISCUSSION TITLE
Does the docker image need all those node modules?

### DISCUSSION THREAD

**Discussion by Jogai** [Lissy93/dashy#403] [General]
The image is 700mb of node_modules. Does dashy need to rebuild when the config changes? If not, it should be possible to only include the build output in the image making it way smaller.

**Comment by lissy93** [comment]
If you can find a way to make it smaller, that would be awesome.
I was a bit reluctant to add the simple-icons package (in #194), which added a lot of size, but there seemed to be demand for it

Once the app is built, it's actually very small, so I agree that it's frustrating that node_modules is so big. But if you don't change your dashboard often, you should be able to build once, and just serve up the dist directory with your own web server

For reference, [there are the sizes of production dependencies](https://bundlephobia.com/scan-results?packages=@formschema/native@2.0.0-beta.5,@sentry/tracing@6.14.3,@sentry/vue@6.13.1,ajv@8.8.1,axios@0.24.0,connect-history-api-fallback@1.6.0,crypto-js@4.1.1,frappe-charts@1.6.2,js-yaml@4.1.0,keycloak-js@15.0.2,register-service-worker@1.6.2,remedial@1.0.8,rsup-progress@2.0.4,simple-icons@5.23.0,v-jsoneditor@1.4.2,v-tooltip@2.1.3,vue@2.6.10,vue-i18n@8.25.1,vue-js-modal@2.0.0-rc.6,vue-json-tree-view@2.1.6,vue-material-tabs@0.1.5,vue-router@3.0.3,vue-select@3.15.0,vue-swatches@2.1.1,vue-toasted@1.1.28,vuex@3.6.2) (about 5mb total), but each package has a bunch of unneccissary stuff like docs, tests, source that's also included when you npm install
